### ISSUE TITLE
Properly handle multi-file sessions

### ISSUE THREAD

**Issue by FlynnOConnell** [MillerBrainObservatory/LBM-CaImAn-MATLAB#18]
TLDR: **High overhead** pre-calculating image sizes from metadata leads to multi-file recordings taking substantially longer proportionally with an increasing number of files.
- 1500 frames in 1 `.tiff` file takes ~2min,
- 1500 frames in 60 `.tiff` files takes ~15min.

## Issue

1. For multi-file recordings in which the user split frames into several `.tiff` files, metadata is pre-calculated from the `.tiff` header and `artist` tag. The overhead associated with loading this metadata is constant across each file and can be preserved for that imaging session.
    - `get_metadata()` will return the same result for `my_raw_file_0001.tif` and `my_raw_file_0002.tif`.
2. Currently, figures are created for each plane. Multi-file recordings now output several figures that overwrite each other.

## Cause
The cause of this issue lies in the loop structure used to iterate through the dataset.

The current logic for iterating through levels of `File`, `Frame`, `Z-Plane` and `ROI` goes as such:
``` text
For every `file` -> load 4D [y,x,z,t] image data % problem child
For every `z-plane`  -> pre-compute image size
For every `ROI` -> Fix Scan Offset
```

## Fix/Refactor

- Persist metadata across each file 
- Limit image output to the first iteration
